lessons/programming/python/file-handling/index.mdx

```diff
@@ -0,0 +1,113 @@
+---
+title: File Handling
+---
+
+<CodeLabLayout title="PythonFileHandling">
+
+# File Handling
+Python has built-in functions to perform operations on files, including reading and writing files.
+
+To perform a file operation, first, a file has to be opened. Then the file operation can be performed. Files are closed afterwards.
+
+## Opening a file
+The built-in **open()** function opens and gaining access to a file. It accesses a file by the specified filename (in the current directory) or a path to that file. 
+A mode can also be specified for different purposes as follows:
+
+**'r'**: Opens the specified file for reading, and this is the default mode of operation. An error occurs if a file does not exist.<br />
+**'w'**: Opens the specified file for writing. A new file will be created if it does not exist.<br />
+**'a'**: Opens the specified file for appending. A new file will be created if it does not exist.<br />
+**'x'**: Creates the specified file. An error occurs if the file exists.<br />
+
+
+As another option, the file could be specified to be one of the two types of mode:
+
+**t**: Text mode, and this is the default mode.<br />
+**b**: Binary mode.<br />
+
+
+## Reading a file
+
+**read()**: read bytes as a string<br />
+**readline()**: read a line in the file as a string<br />
+**readlines()**: read all lines in the file as a list of strings (each line is a string)<br />
+
+## Writing a file
+
+**write(<a string is specified>)**: write a string as a line into the file.<br />
+**writelines(<a list of strings is specified>)**: write multiple strings, each string as a line, into a file.<br />
```

Comment:
Okay @eelay234 so for some reason, using "<" and ">", was causing issues in the mdx file, it was showing errors and the page wasn't loading. Replacing < and > with " and " works.

~~<~~ a string is specified ~~>~~ -> (**"**a string is specified**"**)

~~<~~ a list of strings is specified ~~>~~ -> (**"**a list of strings ~~is~~ -> **are**  specified**"**)